kernel/condvar.c

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) 2020 Intel Corporation.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <kernel.h>
+#include <kernel_structs.h>
+#include <toolchain.h>
+#include <ksched.h>
+#include <wait_q.h>
+#include <syscall_handler.h>
+
+static struct k_spinlock lock;
+
+int z_impl_k_condvar_init(struct k_condvar *condvar)
+{
+	z_waitq_init(&condvar->wait_q);
+	z_object_init(condvar);
+	return 0;
+}
+
+#ifdef CONFIG_USERSPACE
+int z_vrfy_k_condvar_init(struct k_condvar *condvar)
```

Comment:
Coverage shows none of these are being invoked.

Reply:
<img width="636" alt="image" src="https://user-images.githubusercontent.com/180017/104985580-49764980-59df-11eb-8151-d93c5e2f1d6a.png">

Reply:
Confirmed; possibly this was at a point were I didn't realize there were multiple tests so focused the -T argument too tightly.